src/main/java/it/gov/pagopa/payment/service/payment/common/CommonCancelServiceImpl.java

```diff
@@ -49,72 +54,126 @@ public CommonCancelServiceImpl(
-            }
-            log.info("[TRX_STATUS][CANCELLED] The transaction with trxId {} trxCode {}, has been cancelled", trx.getId(), trx.getTrxCode());
-
-            CancelTransactionAuditDTO dto = new CancelTransactionAuditDTO(
-                trx.getInitiativeId(),
-                trx.getId(),
-                trx.getTrxCode(),
-                trx.getUserId(),
-                ObjectUtils.firstNonNull(trx.getRewardCents(), 0L),
-                trx.getRejectionReasons(),
-                merchantId,
-                pointOfSaleId
-            );
-            auditUtilities.logCancelTransaction(dto);
-
-        } catch (RuntimeException e) {
-            auditUtilities.logErrorCancelTransaction(trxId, merchantId);
-            throw e;
-        }
+      log.info("[TRX_STATUS][CANCELLED] The transaction with trxId {} trxCode {}, has been cancelled", trx.getId(), trx.getTrxCode());
+      logCancelTransactionAudit(trx, merchantId, pointOfSaleId);
+
+    } catch (RuntimeException e) {
+      auditUtilities.logErrorCancelTransaction(trxId, merchantId);
+      throw e;
+    }
+  }
+
+  private TransactionInProgress findAndValidateTransaction(String trxId, String merchantId, String acquirerId) {
+    TransactionInProgress trx = repository.findById(trxId)
+        .orElseThrow(() -> new TransactionNotFoundOrExpiredException(
+            "Cannot find transaction with transactionId [%s]".formatted(trxId)));
+
+    if (!trx.getMerchantId().equals(merchantId) || !trx.getAcquirerId().equals(acquirerId)) {
+      throw new MerchantOrAcquirerNotAllowedException(
+          "The merchant with id [%s] associated to the transaction is not equal to the merchant with id [%s]"
+              .formatted(trx.getMerchantId(), merchantId));
+    }
+    return trx;
+  }
+
+  private boolean isDeletableImmediately(TransactionInProgress trx) {
+    return SyncTrxStatus.CREATED.equals(trx.getStatus()) || SyncTrxStatus.IDENTIFIED.equals(trx.getStatus());
+  }
+
+  private void handleAuthorizedTransaction(TransactionInProgress trx, String trxId) {
+    if (cancelExpiration.compareTo(Duration.between(trx.getTrxDate(), OffsetDateTime.now())) < 0) {
+      throw new OperationNotAllowedException(ExceptionCode.PAYMENT_TRANSACTION_EXPIRED,
+          "Cannot cancel expired transaction with transactionId [%s]".formatted(trxId));
+    }
+
+    boolean isReset = trx.getExtendedAuthorization();
+    AuthPaymentDTO refund = rewardCalculatorConnector.cancelTransaction(trx);
+
+    if (refund != null) {
+      trx.setStatus(SyncTrxStatus.CANCELLED);
+      trx.setRewardCents(refund.getRewardCents());
+      trx.setRewards(refund.getRewards());
+      trx.setElaborationDateTime(LocalDateTime.now());
+      sendCancelledTransactionNotification(trx, isReset);
+
+      if (isReset) {
+        trx.setStatus(SyncTrxStatus.CREATED);
```

Comment:
this statement is redundant. The same operation is the first instruction inside resetTransaction()